src/ethereum/osaka/vm/precompiled_contracts/p256verify.py

```diff
@@ -0,0 +1,97 @@
+"""
+Ethereum Virtual Machine (EVM) P256VERIFY PRECOMPILED CONTRACT
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+Introduction
+------------
+Implementation of the P256VERIFY precompiled contract.
+"""
+from ethereum_types.numeric import U256
+
+from ethereum.crypto.elliptic_curve import (
+    SECP256R1A,
+    SECP256R1B,
+    SECP256R1N,
+    SECP256R1P,
+    secp256r1_verify,
+)
+from ethereum.crypto.hash import Hash32
+from ethereum.utils.byte import left_pad_zero_bytes
+
+from ...vm import Evm
+from ...vm.gas import GAS_P256VERIFY, charge_gas
+from ...vm.memory import buffer_read
+
+
+def p256verify(evm: Evm) -> None:
+    """
+    Verifies a P-256 signature.
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+    """
+    data = evm.message.data
+
+    # GAS
+    charge_gas(evm, GAS_P256VERIFY)
+
+    if len(data) != 160:
+        return
+
+    # OPERATION
+    message_hash_bytes = buffer_read(data, U256(0), U256(32))
+    message_hash = Hash32(message_hash_bytes)
+    r = U256.from_be_bytes(buffer_read(data, U256(32), U256(32)))
+    s = U256.from_be_bytes(buffer_read(data, U256(64), U256(32)))
+    qx = U256.from_be_bytes(buffer_read(data, U256(96), U256(32)))
+    qy = U256.from_be_bytes(buffer_read(data, U256(128), U256(32)))
+
+    # Signature component bounds:
+    # Both r and s MUST satisfy 0 < r < n and 0 < s < n
+    r_in_bounds = U256(0) < r < SECP256R1N
+    s_in_bounds = U256(0) < s < SECP256R1N
+    if not (r_in_bounds and s_in_bounds):
+        return
```

Comment:
```suggestion
    if r <= U256(0) or r >= SECP256R1N:
        return
    if s <= U256(0) or s >= SECP256R1N:
        return
```
Small nit, but something like `U256(0) < r < SECP256R1N` is probably very python specific and we want the specs to be more general and readable.